pylinac/fieldparams.py

```diff
@@ -0,0 +1,615 @@
+from .core import image
+from .core.profile import SingleProfile
+from .core import pdf
+from .settings import get_dicom_cmap
+
+interpolate: bool = True
+norm: str = 'max grounded'               # one of 'cax', 'max', 'cax grounded', 'max grounded'
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Parameter functions. Each parameter defined in a protocol list must have an associated parameter function that
+# returns the value of the parameter.
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+def left_edge_50(profile: SingleProfile, *args) -> float:
+    """Return the position of the 50% of max dose value on the left of the profile"""
+    left_edge = abs(profile.distance_to_dose(50, norm, interpolate)[0] - profile.profile_center()[0])/profile.dpmm
+    return left_edge
+
+
+def right_edge_50(profile: SingleProfile, *args):
+    """Return the position of the 50% of max dose value on the right of the profile"""
+    right_edge = abs(profile.distance_to_dose(50, norm, interpolate)[1] - profile.profile_center()[0])/profile.dpmm
+    return right_edge
+
+
+def field_size_50(profile: SingleProfile, *args):
+    """Return the field size at 50% of max dose. Not affected by the normalisation mode.
+    Included for testing purposes"""
+    return profile.fwxm(50)/profile.dpmm
+
+
+def field_size_edge_50(profile: SingleProfile, *args):
+    """Return the field size at 50% of max dose"""
+    return right_edge_50(profile) + left_edge_50(profile)
+
+
+def field_center_fwhm(profile: SingleProfile, *args):
+    """Field center as given by the center of the profile FWHM. Not affected by the normalisation mode.
+    Included for testing purposes"""
+    field_center = (profile.fwxm_center(50, interpolate)[0] - profile.profile_center()[0])/profile.dpmm
+    return field_center
+
+
+def field_center_edge_50(profile: SingleProfile, *args):
+    """Calculates the field center from the 50 dose max field edges. May be different from the field_center_fwxm"""
+    return (right_edge_50(profile) - left_edge_50(profile))/2
+
+
+def penumbra_left_80_20(profile: SingleProfile, *args):
+    """Return the distance between the 80% and 20% max dose values on the left side of the profile"""
+    left_penum = abs(profile.distance_to_dose(80, norm, interpolate)[0]
+                     - profile.distance_to_dose(20, norm, interpolate)[0])/profile.dpmm
+    return left_penum
+
+
+def penumbra_right_80_20(profile: SingleProfile, *args):
+    """Return the distance between the 80% and 20% max dose values on the right side of the profile"""
+    right_penum = abs(profile.distance_to_dose(80, norm, interpolate)[1]
+                      - profile.distance_to_dose(20, norm, interpolate)[1])/profile.dpmm
+    return right_penum
+
+
+def flatness_dose_difference(profile: SingleProfile, ifa: float=0.8):
```

Comment:
What is ifa?

Reply:
In Field Area, or the area over which flatness and symmetry (and other parameters) are calculated. One of the huge problems in comparing different protocols is that each define the IFA differently and the implementation by the manufacturers for the same protocol can differ. The intention is that the function get_infield_area will return the IFA as defined by the specific protocol, but for now I am going with 80%.